source/common/http/conn_manager_impl.h

```diff
@@ -209,6 +209,25 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     void encodeData(Buffer::Instance& data, bool end_stream) override;
     void encodeTrailers(ResponseTrailerMap& trailers) override;
     void encodeMetadata(MetadataMapVector& metadata) override;
+    void setRequestTrailers(Http::RequestTrailerMapPtr&& request_trailers) override {
+      ASSERT(!request_trailers_);
```

Comment:
I assume this ASSERT does not work on the other functions? Can you add comments as to why?